controllers/authControllerLoginController.test.js

```diff
@@ -0,0 +1,223 @@
+import { loginController } from './authController.js';
+import userModel from '../models/userModel.js';
+import authHelper from './../helpers/authHelper.js';
+import JWT from "jsonwebtoken";
+
+// Mock functions
+jest.mock('./../helpers/authHelper', () => ({
+    comparePassword: jest.fn(async (passwordToCheck, correctPassword) => Promise.resolve(true))
+}));
+
+
+jest.mock('../models/userModel.js', () => ({
+    findOne: jest.fn(async (inputQuery) => Promise.resolve({
+        name: "James",
+        email: "james",
+        password: "james123_hashed",
+        phone: "9123 4567",
+        address: "Sentosa",
+        answer: "Badminton"
+    }))
+}));
+
+
+jest.mock('jsonwebtoken', () => ({
+    sign: jest.fn(async (userId, environmentKey, expirationDate) => Promise.resolve("token"))
+}));
+
+
+// Test suite for loginController
+describe('loginController', () => {
+    let res;
+    
+    beforeEach(() => {
+        jest.clearAllMocks();
+
+        //Mock the status and send functions
+        res = {
+            send: jest.fn(),
+            status: jest.fn().mockReturnThis()
+        };
+    });
+
+   
+    describe('should return error messages for its input validations if', () => {
+        it('email is empty', async () => {
```

Comment:
If you are testing whether email is empty, maybe you could put a value in the password so that the test specifically checks that only email is empty

Reply:
Alright sure!